modin/backends/pandas/query_compiler.py

```diff
@@ -213,6 +213,37 @@ def caller(df, *args, **kwargs):
     return caller
 
 
+def handle_readonly_buffer(func: callable, display_name: str = None):
```

Comment:
I have feeling the `query_compiler.py` is getting bloated because of such utilities that may be located in a separate file. Let's continue discussion in #2957 .

Reply:
@YarShev I agree, it can be simplified in some utils. Should we change that in this PR or in the refactor?

Reply:
I suggest going in a separate PR cleaning up the bloat.